### PR TITLE
Add pytests

### PR DESCRIPTION
**Pull request by tr4nt0r**
Main methods are covered with basic cases and also exceptions. Translation and locale related methods are not covered yet. Maybe also more edge cases needed.

### CODE REVIEW

File: .github/workflows/test.yaml
Comment (miaucl) [inline .github/workflows/test.yaml:1]: I would prefer to separate the unit tests and integration tests into two separate workflows. The unit tests are isolated and can run without any secrets and special permissions.

File: requirements_dev.txt [hunk 1]
```diff
@@ -1,8 +1,9 @@
 aiohttp~=3.9
 mypy>=1.8.0
 pre-commit>=3.6.1
+pytest-asyncio==0.23.5
```
Comment (miaucl) [inline requirements_dev.txt:4]: are those explicit or sub-dependencies from pytest?

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+async def session():
+    """Create  a client session."""
+    __session = aiohttp.ClientSession()
+    return __session
+
+
+@pytest.fixture
+async def bring(session):
+    """Create Bring instance."""
+    __bring = Bring(session, os.environ["EMAIL"], os.environ["PASSWORD"])
```
Comment (miaucl) [inline tests/conftest.py:23]: Unit test **must** be isolated, there all responses from `aiohttp` have to be mocked and no real email/password is needed.

Comment (miaucl) [reply tests/conftest.py:23]: Look for `patch` in relation with pytest, it lets you mock individual methods such as aiohttp.get, properties or entire objects such as aiohttp itself.

**Review by miaucl — CHANGES_REQUESTED** [review]

File: requirements_dev.txt (hunk 1 above)
Comment (tr4nt0r) [reply requirements_dev.txt:4]: explicit, required to run async funtions in pytest

File: tests/conftest.py (hunk 2 above)
Comment (tr4nt0r) [reply tests/conftest.py:23]: yeah, babysteps xD
but emulating the bring api to have consistent test data is a good thing for the next iteration. There is also pytest-aiohttp 
 module that does that i think

Comment (miaucl) [reply tests/conftest.py:23]: Cool, that sounds useful!

File: tests/test_bring.py [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+        """Test login with valid user."""
+
+        mocked.post(
+            "https://api.getbring.com/rest/v2/bringauth",
+            status=200,
+            payload=BRING_LOGIN_RESPONSE,
+        )
+
+        async def mocked_get_user_account(*args, **kwargs):
+            """Mock __get_user_account."""
```
Comment (miaucl) [inline tests/test_bring.py:138]: intentional double __?

**Review by miaucl — APPROVED** [review]

File: tests/test_bring.py (hunk 3 above)
Comment (tr4nt0r) [reply tests/test_bring.py]: nope, not intentional, get_user_account is a public method. Fixed it.

File: .github/workflows/pytest.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        # python-version: ["pypy3.9", "pypy3.10", "3.8", "3.9", "3.10", "3.11", "3.12"]
```
Comment (miaucl) [inline .github/workflows/pytest.yaml:12]: this can be removed I guess

File: tests/conftest.py [hunk 5]
```diff
@@ -23,14 +23,47 @@
     "expires_in": 604799,
 }
 
-BRING_USER_ACCOUNT_RESPONSE = {"userLocale": {"language": "de", "country": "DE"}}
+# BRING_USER_ACCOUNT_RESPONSE = {"userLocale": {"language": "de", "country": "DE"}}
```
Comment (miaucl) [inline tests/conftest.py:26]: This can be removed I guess

**Review by miaucl — APPROVED** [review]
